content/millennium/faq/common-issues.md

```diff
@@ -0,0 +1,257 @@
+---
+title: Common Issues
+layout: faq
+---
+
+The topics below are the most common issues that we identify when
+validating apps to run on our platform.  We suggest you work through
+each of these before submitting an app for validation.
+
+# Common Issues
+
+* TOC
+{:toc}
+
+## API Usage
+
+### Avoid Multiple Resource Calls
+
+Ensure that your app is efficient in the way it makes API calls.
+Duplicate API calls should be removed to avoid additional performance
+load on the system and improve application load time. As part of the
+technical review process, we will review logs to see if calls are
+identical each time and if they are retrieving the same data set.
+
+
+### Implement Paging
+
+To be compatible with a wide variety of FHIR services and for future-
+proofing, we require that you implement a paging data retrieval scheme
+as defined by the [HL7 FHIR specification](https://www.hl7.org/fhirhttp.html#paging).
+Your app should handle paging for all resources, even if our FHIR server
+does not support it yet. This ensures your app is compatible with more
+FHIR servers and for ours if we add paging in the future for
+performance.
+
+You can view the default and max page sizes by following the links
+below. Queries on these resources can result in large data sets. If you
+do not implement paging logic, the application can miss relevant
+information because the system will not load more than just the initial
+page. This creates potential for the system to not account for the
+entire patient dataset.  There are two ways we recommend that you test
+your logic once implemented.  First, you can set page size to a lower
+value using the `_count` parameter in the API calls.  Another way is to
+make calls for a patient named Timothy Peters in the Sandbox domain.
+This patient has a very large data set and will often return more than
+the max page size.
```

Comment:
Yes, and as far as I know, we don't provide default sorts that are guaranteed. We probably need to go back to confirm what is and isn't guaranteed, since with paging, it's not as helpful to if there's no sort...